include/arch/arm/arch/64/mode/object/structures.h

```diff
@@ -208,16 +159,10 @@ static inline void *CONST cap_get_archCapPtr(cap_t cap)
         return (void *)(cap_frame_cap_get_capFBasePtr(cap));
 
     case cap_page_table_cap:
-        return PD_PTR(cap_page_table_cap_get_capPTBasePtr(cap));
+        return PT_PTR(cap_page_table_cap_get_capPTBasePtr(cap));
 
-    case cap_page_directory_cap:
-        return PT_PTR(cap_page_directory_cap_get_capPDBasePtr(cap));
-
-    case cap_page_upper_directory_cap:
-        return PUD_PTR(cap_page_upper_directory_cap_get_capPUDBasePtr(cap));
-
-    case cap_page_global_directory_cap:
-        return PGD_PTR(cap_page_global_directory_cap_get_capPGDBasePtr(cap));
+    case cap_vspace_cap:
+        return PT_PTR(cap_vspace_cap_get_capPTBasePtr(cap));
```

Comment:
we're returning a `void*` in the end, so this probably doesn't matter much, but should this be `VSPACE_PTR` instead of `PT_PTR` here?

```suggestion
    case cap_vspace_cap:
        return VSPACE_PTR(cap_vspace_cap_get_capPTBasePtr(cap));
```